### ISSUE TITLE
Inconvenient way of handling lists in FlatSource

### ISSUE THREAD

**Issue by NorthFury** [uchuhimo/konf#6]
When using something like `.withSourceFrom.env()` or `.withSourceFrom.systemProperties()` for an `OptionalItem<List<String>>` for example, the current implementation expects multiple environment variables defined like this: `MY_KEY_0`, `MY_KEY_1` and so on. This is quite inconvenient.

It would be nice to parse `MY_KEY` either as a JSON or comma separated values.

**Comment by uchuhimo** [comment]
Thank you for your suggestions. I can add support for parsing comma separated values as list in flat source (properties file, XML, etc.) when I have time off, or your contributions are welcome if you have interest. Further support of complex hierarchical structure in flat source (e.g. JSON-like structure) is not in my plan (at least for now). I prefer to migrate to hierarchical source (JSON, YAML, etc.) if I need complex hierarchical structure.

**Comment by NorthFury** [comment]
Comma separated values would be more than enough. For now I extended FlatSource in my project. I'm not satisfied with my solution but it fits my needs for now. I'll see if I can find the time for a proper solution.

**Comment by NorthFury** [comment]
I will be using my hack and won't create a PR for this. Until this gets added here is a workaround but only if you require lists of string. It will fail for lists of any other types.

```kotlin
class CsvStringListFlatSource(
        map: Map<String, String>,
        prefix: String = "",
        type: String = "",
        context: Map<String, String> = mapOf()
) : FlatSource(map, prefix, type, context) {

    override fun getOrNull(path: Path): Source? {
        if (path.isEmpty()) return this

        if (!contains(path))
            return null

        if (prefix.isEmpty()) {
            return CsvStringListFlatSource(map, path.name, context = context)
        } else {
            return CsvStringListFlatSource(map, "$prefix.${path.name}", context = context)
        }
    }

    override fun toList(): List<Source> {
        return this.map[this.prefix]
                ?.split(',')
                ?.map { ValueSource(it) }
                ?: emptyList()
    }
}
```

**Comment by uchuhimo** [comment]
Support for parsing comma separated values as list in the flat source have been released in [v0.10](https://github.com/uchuhimo/konf/releases/tag/v0.10). Feedback is welcome if any bug is found.
